source/simulator/assemblers/interface.cc

```diff
@@ -53,8 +53,9 @@ namespace aspect
           grads_phi_u (stokes_dofs_per_cell, numbers::signaling_nan<SymmetricTensor<2,dim>>()),
           div_phi_u (stokes_dofs_per_cell, numbers::signaling_nan<double>()),
           phi_p (stokes_dofs_per_cell, numbers::signaling_nan<double>()),
+          phi_u(stokes_dofs_per_cell,numbers::signaling_nan<Tensor<1,dim>>()),
           phi_p_c (add_compaction_pressure ? stokes_dofs_per_cell : 0, numbers::signaling_nan<double>()),
-          grad_phi_p (add_compaction_pressure ? stokes_dofs_per_cell : 0, numbers::signaling_nan<Tensor<1,dim>>()),
+          grad_phi_p (stokes_dofs_per_cell, numbers::signaling_nan<Tensor<1,dim>>()),
```

Comment:
```suggestion
          grad_phi_p ((add_compaction_pressure || use_bfbt) ? stokes_dofs_per_cell : 0, numbers::signaling_nan<Tensor<1,dim>>()),
```
but you need to pass use_bfbt to the constructor.